### ISSUE TITLE
Unexplained "dryRun is not supported" error for deploying Application targetted at OCM cluster

### ISSUE THREAD

**Issue by vyta** [kubevela/kubevela#5361]
**Describe the bug**

Application workflow fails with "cannot create object: dryRun is not supported)" with OCM managedclusters

**To Reproduce**

1. Create hub cluster w/ vela + OCM hub + gateway manager
2. Create member cluster and join 
3. Create Application definition with membercluster listed in topology
4. Observe application workflow fail with:
  ```
  Message: run step(provider=multicluster,do=deploy): Found 1 errors. [(error encountered in cluster cluster1: Dispatch: pre-dispatch dryrun failed: Found 1 errors. [(cannot create object: dryRun is not supported)])]
 ```

**Expected behavior**

Should detect cluster and deploy application

**KubeVela Version**

CLI Version: v1.7.0
Core Version: 1.7.0

**Cluster information**

1.24.0

**Comment by Somefive** [comment]
Could you see what cluster-gateway image version are you using? You could run `kubectl get deployment -n vela-system -owide` and see it.

**Comment by vyta** [comment]
> Could you see what cluster-gateway image version are you using? You could run `kubectl get deployment -n vela-system -owide` and see it.

v1.7.0-alpha.3

**Comment by Somefive** [comment]
I guess it might be because the OCM cluster proxy does not support the dryrun request proxy. Could you try to add `--set featureGates.preDispatchDryRun=false` to the `helm install` process. It will let the KubeVela controller skip the dryrun phase for dispatching resources.

**Comment by testinfected** [comment]
We're still getting that image with Vela 1.9.9

Our cluster gateway is 1.9.0-alpha2. Any idea where to look to understand/fix this issue?

Thanks

**Comment by testinfected** [comment]
We saw that Cluster Gateway 1.9.0 was released, but that does not solve the issue.

Would any of the Kubela maintainers ( @FogDong ? @wonderflow ? @Somefive ?) please give a clue on how to solve this issue. 

Thanks guys,

**Comment by FogDong** [comment]
Hi, have you tried  add `--set featureGates.preDispatchDryRun=false` to the helm install process?

**Comment by testinfected** [comment]
Hey @FogDong yes that works although we'd like to keep the pre-dispatch dry run if possible. Are you telling me it's not compatible with multicluster deployment?

**Comment by FogDong** [comment]
Yeah I think right now OCM cluster proxy still does not support the dryrun request proxy. Any insights from @Somefive ?

**Comment by FogDong** [comment]
In a standard cluster setup, the `dryRun` option should indeed function as expected. I will investigate this issue further if it is not working as expected.

**Comment by testinfected** [comment]
@FogDong it's not working correctly for us. We use normal `vela join` and don't have any OCM addons installed.

We have to disable dry runs unfortunately. Any clues on what could be the problem here?

**Comment by testinfected** [comment]
Bump, any idea @FogDong? Thanks in advance
